File: services/aas-designer/AasDesignerModel/Model/OrgaRechnung.cs
namespace AasDesignerApi.Model
{
    public class OrgaRechnung : IMetadata, IHardDeletable, IProtokollierbar
    {
        public long Id { get; set; }

        public string Daten { get; set; } = string.Empty; // als JsonObjekt?
        public double Summe { get; set; }
        public DateTime Rechnungsdatum { get; set; }

        public Rechnungsmonat Rechnungsmonat { get; set; } = null!;
        public long RechnungsmonatId { get; set; }
        public Organisation Organisation { get; set; } = null!;
        public long OrganisationId { get; set; }

        #region Verwaltungsdaten
        public DateTime AnlageDatum { get; set; } = DateTime.Now;
        public string AnlageBenutzer { get; set; } = string.Empty;
        public DateTime AenderungsDatum { get; set; } = DateTime.Now;
        public string AenderungsBenutzer { get; set; } = string.Empty;
        public bool Geloescht { get; set; } = false;
        public int AenderungsZaehler { get; set; } = 0;

        #endregion
    }
}

File: frontend/libs/aas-designer-model/src/lib/orga-rechnung.ts
import { Organisation } from './organisation';

export class OrgaRechnung {
  id: number | undefined;
  rechnungsdatum: Date | null = null;
  summe: number = 0;
  daten: string = '';

  anlageDatum: Date | null = null;
  aenderungsDatum: Date | null = null;

  organisation: Organisation | null = null;

  static fromDto(dto: any) {
    const orgaRechnung = new OrgaRechnung();
    Object.assign(orgaRechnung, dto);

    if (dto.anlageDatum != null) orgaRechnung.anlageDatum = new Date(dto.anlageDatum);
    if (dto.aenderungsDatum != null) orgaRechnung.aenderungsDatum = new Date(dto.aenderungsDatum);
    if (dto.rechnungsdatum != null) orgaRechnung.rechnungsdatum = new Date(dto.rechnungsdatum);

    if (dto.organisation != null) orgaRechnung.organisation = Organisation.fromDto(dto.organisation);

    return orgaRechnung;
  }
}
